Sources/SwiftDocC/Utility/MarkupExtensions/ListItemExtractor.swift

```diff
@@ -710,3 +229,201 @@ struct TaggedListItemExtractor: MarkupRewriter {
+    /// The range of the raw tag text
+    var tagRange: SourceRange?
+    /// The complete content related to this tag
+    var contents: [Markup]
+    /// The range of the tag and its content
+    var range: SourceRange?
+    
+    init(rawTag: String, tagRange: SourceRange?, contents: [Markup], range: SourceRange?) {
+        self.rawTag = rawTag
+        self.knownTag = .init(rawTag)
+        self.tagRange = tagRange
+        self.contents = contents
+        self.range = range
+    }
+    
+    enum KnownTag {
+        case returns
+        case `throws`
+        case parameter(String)
+        case parameters
+        
+        case dictionaryKey(String)
+        case dictionaryKeys
+        
+        case httpBody
+        case httpResponse(String)
+        case httpResponses
+        case httpParameter(String)
+        case httpParameters
+        case httpBodyParameter(String)
+        case httpBodyParameters
+        
+        init?(_ string: String) {
+            let components = string.components(separatedBy: .whitespaces).filter { !$0.isEmpty }
+            
+            switch components.first?.lowercased() {
+            case "returns":
+                self = .returns
+            case "throws":
+                self = .throws
+            case "parameter" where components.count == 2:
+                self = .parameter(String(components.last!))
+            case "parameters":
+                self = .parameters
+            case "dictionarykey" where components.count == 2:
+                self = .dictionaryKey(String(components.last!))
+            case "dictionarykeys":
+                self = .dictionaryKeys
+            case "httpbody":
+                self = .httpBody
+            case "httpresponse" where components.count == 2:
+                self = .httpResponse(String(components.last!))
+            case "httpresponses":
+                self = .httpResponses
+            case "httpparameter" where components.count == 2:
+                self = .httpParameter(String(components.last!))
+            case "httpparameters":
+                self = .httpParameters
+            case "httpbodyparameter" where components.count == 2:
+                self = .httpBodyParameter(String(components.last!))
+            case "httpbodyparameters":
+                self = .httpBodyParameters
+            default:
+                return nil
+            }
+        }
+    }
+}
+
+private extension ListItem {
+    func extractTag() -> ExtractedTag? {
+        guard childCount > 0,
+              let paragraph = child(at: 0) as? Paragraph,
+              let (name, nameRange, remainderOfFirstParagraph) = paragraph.inlineChildren.splitNameAndContent()
+        else {
+            return nil
+        }
+        
+        return ExtractedTag(rawTag: name, tagRange: nameRange, contents: remainderOfFirstParagraph + children.dropFirst(), range: range)
+    }
+    
+    func extractInnerTagOutline() -> [ExtractedTag] {
```

Comment:
Nit: When I was reading this function's callsite, it wasn't clear to me from the name what it does. Maybe you could add a comment/example like you have elsewhere showing what "tag outline" means?

Reply:
Added in https://github.com/swiftlang/swift-docc/pull/1014/commits/427bff2fe1e02e9a79f7b57f7ae95302325d6f42